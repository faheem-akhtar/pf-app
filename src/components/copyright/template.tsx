import { useTranslation } from 'helpers/translation/hook';

export const CopyrightTemplate: React.FunctionComponent = () => {
  const { t } = useTranslation();
  return <div>{t('copyright')}</div>;
};
