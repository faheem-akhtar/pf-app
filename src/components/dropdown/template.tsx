import { ChangeEvent } from 'react';

import { domClassMerge } from 'helpers/dom/class-merge';

import { DropdownTemplatePropsInterface } from './template-props.interface';
import { FiltersValueFieldChoiceInterface } from 'components/filters/value/field/choice.interface';

import styles from './dropdown.module.scss';
import { useTranslation } from 'next-i18next';

// TODO-FE[TPNX-3056] remove this file
export const DropdownTemplate = <T extends unknown>({
  title,
  value,
  choices,
  onChange,
  className,
}: DropdownTemplatePropsInterface<T>): JSX.Element => {
  const { t } = useTranslation();
  return (
    <div className={domClassMerge(className, styles.container)}>
      {title && <span className={styles.title}>{title}</span>}
      <select
        value={String(choices.indexOf(choices.find((c) => c.value === value) as FiltersValueFieldChoiceInterface<T>))}
        onChange={(e): void =>
          onChange(choices[parseInt((e as unknown as ChangeEvent<HTMLSelectElement>).target.value, 10)].value)
        }
      >
        {choices.map((choice, index) => (
          <option key={index} value={index}>
            {choice.label ? choice.label : t('Any')}
          </option>
        ))}
      </select>
    </div>
  );
};
