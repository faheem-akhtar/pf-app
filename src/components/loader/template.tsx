import { domClassMerge } from 'helpers/dom/class-merge';

import styles from './loader.module.scss';
import { LoaderTemplatePropsInterface } from './template-props.interface';

// TODO-FE[TPNX-2931] move it to the library, it's used in Button
export const LoaderTemplate = (props: LoaderTemplatePropsInterface): JSX.Element => (
  <div className={domClassMerge(styles.dropdownLoader, props.containerCssClass)}>
    <div className={styles.container}>
      <div className={domClassMerge(styles.dot, { [styles.inverted]: !!props.isInverted }, props.dotCssClass)} />
      <div
        className={domClassMerge(
          styles.dot,
          { [styles.inverted]: !!props.isInverted },
          styles.middle,
          props.dotCssClass
        )}
      />
      <div
        className={domClassMerge(styles.dot, { [styles.inverted]: !!props.isInverted }, styles.last, props.dotCssClass)}
      />
    </div>
  </div>
);
