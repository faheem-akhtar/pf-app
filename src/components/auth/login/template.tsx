import React, { FunctionComponent } from 'react';

import { AuthLoaderComponent } from 'components/auth/loader/component';
import { AuthLoginFieldEnum } from 'components/auth/login/field.enum';
import { ErrorMessageComponent } from 'components/error-message/component';
import { IconFacebookTemplate } from 'components/icon/facebook-template';
import { IconGoogleTemplate } from 'components/icon/google-template';
import { ReCaptchaComponent } from 'components/re-captcha/component';
import { domClassMerge } from 'helpers/dom/class-merge';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonIconPositionEnum } from 'library/button/icon-position.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { InputBaseComponent } from 'library/input/base/component';

import styles from '../auth.module.scss';
import { AuthLoginTemplatePropsInterface } from './template-props.interface';

export const AuthLoginTemplate: FunctionComponent<AuthLoginTemplatePropsInterface> = (props) => (
  <div data-testid='AuthLoginComponent'>
    <h1 className={styles.heading}>{props.t('sign-in')}</h1>
    {/* Facebook */}
    <ButtonTemplate
      type='button'
      className={domClassMerge(styles.button, styles.facebook)}
      componentType={ButtonComponentTypeEnum.tertiary}
      size={ButtonSizeEnum.small}
      onClick={props.onFacebookLoginClick}
      icon={{ component: IconFacebookTemplate, position: ButtonIconPositionEnum.left }}
    >
      {props.t('auth/sign-in-facebook')}
    </ButtonTemplate>
    {/* Google */}
    <ButtonTemplate
      type='button'
      className={domClassMerge(styles.button, styles.google)}
      componentType={ButtonComponentTypeEnum.tertiary}
      size={ButtonSizeEnum.small}
      onClick={props.onGoogleLoginClick}
      icon={{ component: IconGoogleTemplate, position: ButtonIconPositionEnum.left }}
    >
      {props.t('auth/sign-in-google')}
    </ButtonTemplate>
    <div className={styles.divider}>
      <span className={styles.text}>{props.t('auth/or')}</span>
    </div>
    <AuthLoaderComponent isEnabled={props.isLoading} isCentered={true} />
    {props.errorMessage && <ErrorMessageComponent message={props.errorMessage} />}
    <form onSubmit={props.onFormSubmit}>
      <div className={styles['input-area']}>
        <InputBaseComponent
          id='email'
          name='email'
          placeholder={props.t('email')}
          type='email'
          value={props.email}
          error={!!props.errors[AuthLoginFieldEnum.email]}
          onChange={props.onEmailChange}
          errorText={props.errors[AuthLoginFieldEnum.email]}
        />
      </div>
      <div className={styles['input-area']}>
        <InputBaseComponent
          id='password'
          name='password'
          placeholder={props.t('password')}
          type='password'
          value={props.password}
          error={!!props.errors[AuthLoginFieldEnum.password]}
          onChange={props.onPasswordChange}
          errorText={props.errors[AuthLoginFieldEnum.password]}
        />
      </div>
      <div className={styles['input-area']} onClick={props.onForgotPassword}>
        <div className={domClassMerge(styles.link, styles['forgot-password'])}>{props.t('auth/forgot-password')}?</div>
      </div>
      <div className={styles['input-area']}>
        <ButtonTemplate
          className={styles.button}
          type='submit'
          componentType={ButtonComponentTypeEnum.primary}
          size={ButtonSizeEnum.regular}
        >
          {props.t('log-in')}
        </ButtonTemplate>
      </div>
      <ReCaptchaComponent />
    </form>
    <div className={domClassMerge(styles.link, styles['create-account'])} onClick={props.onRegister}>
      {props.t('auth/not-registered-yet')}?&nbsp;{props.t('auth/create-account')}
    </div>
  </div>
);
