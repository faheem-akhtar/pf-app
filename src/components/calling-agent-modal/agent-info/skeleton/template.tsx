import { SkeletonTemplate } from 'library/skeleton/template';

import styles from './calling-agent-modal-agent-info-skeleton.module.scss';

export const CallingAgentModalAgentInfoSkeletonTemplate = (): JSX.Element => (
  <div className={styles.container}>
    <SkeletonTemplate width='7.04rem' height='7.04rem' class={styles['skeleton--image']} />

    <div className={styles.content}>
      <SkeletonTemplate width='8rem' height='2.4rem' class={styles.skeleton} />
      <SkeletonTemplate width='100%' height='2.5rem' class={styles.skeleton} />
    </div>
  </div>
);
