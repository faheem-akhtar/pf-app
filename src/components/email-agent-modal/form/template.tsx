import { configEmailAgentMustAcceptTerms } from 'config/email-agent/must-accept-terms';
import { configEmailAgentReceiveAdsEnabled } from 'config/email-agent/receive-ads-enabled';

import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { EmailAgentModalFormTemplatePropsInterface } from './template-props.interface';
import { EmailAgentModalFormWidgetAcceptTermsComponent } from './widget/accept-terms-component';
import { EmailAgentModalFormWidgetEmailAlertComponent } from './widget/email-alert-component';
import { EmailAgentModalFormWidgetEmailComponent } from './widget/email-component';
import { EmailAgentModalFormWidgetMessageComponent } from './widget/message-component';
import { EmailAgentModalFormWidgetNameComponent } from './widget/name-component';
import { EmailAgentModalFormWidgetPhoneComponent } from './widget/phone-component';
import { EmailAgentModalFormWidgetReceiveAdvertisingComponent } from './widget/receive-advertising-component';

import styles from './email-agent-modal-form.module.scss';

export const EmailAgentModalFormTemplate = ({
  fieldsValue,
  setFieldsValue,
  onSubmit,
  t,
}: EmailAgentModalFormTemplatePropsInterface): JSX.Element => {
  return (
    <form className={styles.form} onSubmit={onSubmit}>
      <EmailAgentModalFormWidgetNameComponent
        t={t}
        value={fieldsValue.name}
        onChange={(newValue): void => setFieldsValue({ ...fieldsValue, name: newValue })}
      />
      <EmailAgentModalFormWidgetEmailComponent
        t={t}
        value={fieldsValue.email}
        onChange={(newValue): void => setFieldsValue({ ...fieldsValue, email: newValue })}
      />
      <EmailAgentModalFormWidgetPhoneComponent
        t={t}
        value={fieldsValue.phone}
        onChange={(newValue): void => setFieldsValue({ ...fieldsValue, phone: newValue })}
      />
      <EmailAgentModalFormWidgetMessageComponent
        t={t}
        value={fieldsValue.message}
        onChange={(newValue): void => setFieldsValue({ ...fieldsValue, message: newValue })}
      />
      <EmailAgentModalFormWidgetEmailAlertComponent
        t={t}
        value={fieldsValue.emailAlert}
        onChange={(newValue): void => setFieldsValue({ ...fieldsValue, emailAlert: newValue })}
      />
      {configEmailAgentMustAcceptTerms && (
        <EmailAgentModalFormWidgetAcceptTermsComponent
          t={t}
          value={fieldsValue.acceptTerms}
          onChange={(newValue): void => setFieldsValue({ ...fieldsValue, acceptTerms: newValue })}
        />
      )}
      {configEmailAgentReceiveAdsEnabled && (
        <EmailAgentModalFormWidgetReceiveAdvertisingComponent
          t={t}
          value={fieldsValue.receiveAdvertising}
          onChange={(newValue): void => setFieldsValue({ ...fieldsValue, receiveAdvertising: newValue })}
        />
      )}

      <ButtonTemplate
        type='submit'
        size={ButtonSizeEnum.regular}
        componentType={ButtonComponentTypeEnum.primary}
        className={styles.button}
      >
        {t('agent-modal/cta-send-message')}
      </ButtonTemplate>
    </form>
  );
};
