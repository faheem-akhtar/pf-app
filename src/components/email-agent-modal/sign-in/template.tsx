import { IconSolidEnvelopeOpenTemplate } from 'components/icon/solid/envelope-open-template';
import { IconSolidFavoriteTemplate } from 'components/icon/solid/favorite-template';
import { IconThinSyncTemplate } from 'components/icon/thin/sync-template';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';

import styles from './email-agent-modal-sign-in.module.scss';
import { EmailAgentModalSignInItemTemplate } from './item-template';
import { EmailAgentModalSignInTemplatePropsInterface } from './template-props.interface';

export const EmailAgentModalSignInTemplate = ({
  t,
  closeModal,
  openAuthRef,
}: EmailAgentModalSignInTemplatePropsInterface): JSX.Element => (
  <div className={styles.container}>
    <p className={styles.title}>{t('agent-modal/sign-in-title')}</p>

    <ul className={styles.list}>
      <EmailAgentModalSignInItemTemplate icon={<IconSolidFavoriteTemplate class={styles.icon} clipped />}>
        {t('agent-modal/sign-in/saved-properties-label')}
      </EmailAgentModalSignInItemTemplate>

      <EmailAgentModalSignInItemTemplate icon={<IconSolidEnvelopeOpenTemplate class={styles.icon} />}>
        {t('agent-modal/sign-in/agent-communication-label')}
      </EmailAgentModalSignInItemTemplate>

      <EmailAgentModalSignInItemTemplate icon={<IconThinSyncTemplate class={styles.icon} />}>
        {t('agent-modal/sign-in/sync-all-devices-label')}
      </EmailAgentModalSignInItemTemplate>
    </ul>

    <ButtonTemplate
      type='button'
      size={ButtonSizeEnum.regular}
      componentType={ButtonComponentTypeEnum.primary}
      onClick={(): void => {
        closeModal();
        openAuthRef.current();
      }}
      fullWidth
    >
      {t('sign-in')}
    </ButtonTemplate>

    <button className={styles.notNowButton} onClick={closeModal}>
      {t('not-now')}
    </button>
  </div>
);
