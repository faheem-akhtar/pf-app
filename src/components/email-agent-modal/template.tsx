import { IconThickSmallCloseTemplate } from 'components/icon/thick/small-close-template';

import { EmailAgentModalFormErrorMessageTemplate } from './form/error-message-template';
import { EmailAgentModalFormSuccessTemplate } from './form/success-template';
import { EmailAgentModalFormTemplate } from './form/template';
import { EmailAgentModalStatusEnum } from './status.enum';
import { EmailAgentModalTemplatePropsInterface } from './template-props.interface';

import styles from './email-agent-modal.module.scss';

export const EmailAgentModalTemplate = (props: EmailAgentModalTemplatePropsInterface): JSX.Element => {
  const body =
    props.status === EmailAgentModalStatusEnum.submitted ? (
      <EmailAgentModalFormSuccessTemplate closeModal={props.closeModal} t={props.t} />
    ) : (
      <>
        <h2 className={styles.name}>{props.propertyName}</h2>
        <EmailAgentModalFormTemplate
          fieldsValue={props.fieldsValue}
          setFieldsValue={props.setFieldsValue}
          onSubmit={props.onSubmit}
          t={props.t}
          errors={props.errors}
          loading={props.loading}
        />
      </>
    );

  return (
    <div
      className={styles.container}
      onClick={(e): void => {
        e.stopPropagation();
      }}
    >
      <header className={styles.header}>
        <h1 className={styles.title}>{props.t('agent-modal/email-form-title')}</h1>
        <div className={styles.closeButton} onClick={props.closeModal}>
          <IconThickSmallCloseTemplate class={styles.closeIcon} clipped />
        </div>
      </header>

      {props.error && <EmailAgentModalFormErrorMessageTemplate error={props.error} />}
      {body}
    </div>
  );
};
