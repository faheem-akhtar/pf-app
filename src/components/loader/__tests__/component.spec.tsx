/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react';

import { LoaderTemplate } from '../template';

describe('LoaderTemplate', () => {
  it('renders without throwing any errors', () => {
    const { container } = render(<LoaderTemplate />);

    expect(container).toMatchSnapshot();
  });
});
