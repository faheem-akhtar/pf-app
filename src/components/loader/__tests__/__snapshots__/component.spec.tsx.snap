// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderTemplate renders without throwing any errors 1`] = `
<div>
  <div
    class="dropdownLoader"
    data-testid="loader-template"
  >
    <div
      class="container"
    >
      <div
        class="dot"
      />
      <div
        class="dot middle"
      />
      <div
        class="dot last"
      />
    </div>
  </div>
</div>
`;
